### DISCUSSION TITLE
Change default postgres password

### DISCUSSION THREAD

**Discussion by ErlandssonA** [plausible/analytics#3861] [Self-Hosted Support]
Hello! I installed Plausible self hosted yesterday using both the install documentation and a YouTube video. In the video he mentioned that you should change the postgres password in the `docker-compose.yml`. How can i do that? If i just change it in that file it cant connect because the password is wrong.

So i guess I somehow need to change the password and then also change it in the `docker-compose.yml`.

**Comment by ruslandoga** [comment]
👋 @ErlandssonA 

To change the password for the PostgreSQL database, you set a custom [`POSTGRES_PASSWORD`](https://github.com/plausible/community-edition/blob/0a9c71320a7cd8d0c1b71e9ec2611d04cf1f9bba/docker-compose.yml#L15) on `plausible_db` container and in `DATABASE_URL` on `plausible` container. Please see https://github.com/plausible/community-edition/tree/v2.0.0#configure for more information on the configuration options.

**Reply by ruslandoga** [reply]
Note that if you start a container with one set of ENV vars and then update the ENV vars and restart the container, they won't take effect due to the immutable nature of the containers. The container needs to be recreated.

**Reply by ErlandssonA** [reply]
Okay, and the different containers you are talking about. Im very new to docker, so I dont really know this stuff...

Would you mind walking me through it?

**Reply by ruslandoga** [reply]
The containers in the default setup are listed in https://github.com/plausible/community-edition/blob/v2.0.0/docker-compose.yml

To learn more about Docker you can read their [documentation](https://docs.docker.com) or ask [ChatGPT,](http://chat.openai.com/) it's really good at this sort of thing.

**Reply by ErlandssonA** [reply]
I ended up just reinstalling the whole thing...

Now im just using the default password, is it a bad idea to do that?

**Reply by ruslandoga** [reply]
If the database is not exposed to the internet, it doesn't matter much.

**Reply by ErlandssonA** [reply]
Alright, and will it be exposed if i just use the "default" installation?

**Reply by ruslandoga** [reply]
No, it won't be since no `ports:` are "published" on the `plausible_db` container.

**Reply by ErlandssonA** [reply]
Alright, thank you very much!
